index.js

```diff
@@ -131,6 +131,36 @@ class ScmBase {
     stats() {
         return {};
     }
+
+    /**
+     * Return a unique identifier for the scmUrl
+     * @method getRepoId
+     * @param  {Object}    config        Configuration
+     * @param  {String}    config.scmUrl The scmUrl to generate ID
+     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    getRepoId(config) {
+        const result = Joi.validate(config, dataSchema.plugins.scm.getRepoId);
```

Comment:
`getRepoId` is what will be called first to do validation. The actual implementation is inside `_getRepoId`, which is implemented in https://github.com/screwdriver-cd/scm-github/pull/8.